src/native/minipal/errno.h

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
This file should not be needed.

CoreCLR PAL implementation should be able to use definition of these constants from Haiku system headers. I assume that all these constants are defined in Haiky system headers. Is that correct?

Components using the CoreCLR PAL should use definitions provided by the CoreCLR PAL header.

Reply:
I believe the discussion has been done here: https://github.com/dotnet/runtime/issues/55803#issuecomment-886096255, unless something major has changed since the .NET 6 - 7 times.

@jessicah's port (https://github.com/jessicah/dotnet-runtime) uses `#ifdef` for every `errno` wherever needed; this header gathers them all to one place.

Reply:
I agree that there may be some changes required to deal with the errno given how the CoreCLR PAL is structured (https://github.com/dotnet/runtime/issues/55803#issuecomment-886696103).

All changes related to errno should be under src/coreclr/pal. No changes should be needed outside this directory, except of deleting unnecessary code. I have left some comments on the @jessicah's commit: https://github.com/jessicah/dotnet-runtime/commit/7bd4f1d40b4d938b81eecbc77d80e145e005118b

Reply:
Done. I've removed the `minpal/errno.h` header.

However, unlike what you said about the definitions in `sstring.cpp` (a file _outside_ `src/coreclr/pal`, located in `src/coreclr/utilcode` instead), the definitions are still needed:

https://github.com/dotnet/runtime/blob/b09e18e8b6092e35611c2567bd55e8909ea78d7d/src/coreclr/utilcode/sstring.cpp#L1794-L1803

For these files, I've replaced the `#include <minipal/errno.h>` with `#include <pal.h>`, though I don't really know what's the correct solution.

For now, nothing seems to have broken. Builds are still fine, `paltests` still passing.

Reply:
Oops, seems like `pal.h` is not accessible from `sstring.cpp`:

> D:\a\_work\1\s\src\coreclr\utilcode\sstring.cpp(14): fatal error C1083: Cannot open include file: 'pal.h': No such file or directory

Reply:
@trungnt2910 the pal.h is not available on Windows. You should never include it anywhere from the runtime code explicitly.  It gets automatically included when runtime includes any of the standard headers. They are redirected to palrt.h and that one pulls in pal.h